test/e2e/seamless_migration_test.go

```diff
@@ -0,0 +1,234 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/pkg/util/names"
+	"github.com/karmada-io/karmada/test/e2e/framework"
+	"github.com/karmada-io/karmada/test/helper"
+)
+
+var _ = ginkgo.Describe("Seamless migration testing", func() {
+	var member1 string
+	var member1Client kubernetes.Interface
+
+	ginkgo.BeforeEach(func() {
+		member1 = "member1"
+		member1Client = framework.GetClusterClient(member1)
+	})
+
+	// referring to related manual steps: https://github.com/karmada-io/karmada/pull/3821#issuecomment-1649238940
+	ginkgo.Context("Test migrate namespaced resource: Deployment", func() {
+		var deployment *appsv1.Deployment
+		var propagationPolicy *policyv1alpha1.PropagationPolicy
+		var bindingName string
+
+		ginkgo.BeforeEach(func() {
+			deployment = helper.NewDeployment(testNamespace, deploymentNamePrefix+rand.String(RandomStrLength))
+			propagationPolicy = helper.NewPropagationPolicy(deployment.Namespace, deployment.Name, []policyv1alpha1.ResourceSelector{
+				{
+					APIVersion: deployment.APIVersion,
+					Kind:       deployment.Kind,
+					Name:       deployment.Name,
+				},
+			}, policyv1alpha1.Placement{
+				ClusterAffinity: &policyv1alpha1.ClusterAffinity{ClusterNames: []string{member1}},
+			})
+			bindingName = names.GenerateBindingName(deployment.Kind, deployment.Name)
+		})
+
+		ginkgo.BeforeEach(func() {
+			// Create Deployment in member1 cluster
+			framework.CreateDeployment(member1Client, deployment)
+			// Create Deployment in karmada control plane
+			framework.CreateDeployment(kubeClient, deployment)
+			// Create PropagationPolicy in karmada control plane without conflictResolution field
+			framework.CreatePropagationPolicy(karmadaClient, propagationPolicy)
+
+			ginkgo.DeferCleanup(func() {
+				// Delete Deployment in karmada control plane
+				framework.RemoveDeployment(kubeClient, deployment.Namespace, deployment.Name)
+				// Delete PropagationPolicy in karmada control plane
+				framework.RemovePropagationPolicy(karmadaClient, propagationPolicy.Namespace, propagationPolicy.Name)
+
+				// Verify Deployment in member cluster will be deleted automatically after promotion since it be deleted from Karmada
```

Comment:
```suggestion
				// Verify Deployment in member cluster will be deleted automatically after promotion since it has been deleted from Karmada
```

Reply:
You are absolutely right, thank you very much for your advice~